src/wrappers/OsipiBase.py

```diff
@@ -0,0 +1,210 @@
+        
+        # If the user inputs an algorithm to OsipiBase, it is intereprete as initiating
+        # an algorithm object with that name.
+        if algorithm:
+            self.osipi_initiate_algorithm(algorithm, bvalues=bvalues, thresholds=thresholds, bounds=bounds, initial_guess=initial_guess)
+            
+    def osipi_initiate_algorithm(self, algorithm, **kwargs):
+        """Turns the class into a specified one by the input.
+        This method can be used instead of specifically importing that class and initiating it.
+        
+        WIP: Add args and kwargs!
+
+        Args:
+            algorithm (string): The name of the algorithm, should be the same as the file in the src/standardized folder without the .py extension.
+        """
+        
+        # Import the algorithm
+        import_base_path = "src.standardized"
+        import_path = import_base_path + "." + algorithm
+        #Algorithm = getattr(importlib.import_module(import_path), algorithm)
+        # Change the class from OsipiBase to the specified algorithm
+        self.__class__ = getattr(importlib.import_module(import_path), algorithm)
+        self.__init__(**kwargs)
+
+    def osipi_fit(self, data=None, bvalues=None, thresholds=None, bounds=None, initial_guess=None, **kwargs):
+        """Fits the data with the bvalues
+        Returns [S0, f, Dstar, D]
+        """
+        
+        # We should first check whether the attributes in the __init__ are not None
+        # Then check if they are input here, if they are, these should overwrite the attributes
+        use_bvalues = bvalues if bvalues is not None else self.bvalues
+        kwargs["bvalues"] = use_bvalues
+        use_thresholds = thresholds if self.bvalues is None else self.thresholds
+        use_bounds = bounds if self.bounds is None else self.bounds
+        use_initial_guess = initial_guess if self.initial_guess is None else self.initial_guess
+        
+        #args = [data, use_bvalues, use_thresholds]
+        args = [data, use_thresholds]
+        if self.required_bounds or self.required_bounds_optional:
+            args.append(use_bounds)
+        if self.required_initial_guess or self.required_initial_guess_optional:
+            args.append(use_initial_guess)
+            
+        # Run a check_requirements method that makes sure that these inputs fulfil the requirements
+        
+        
+        # Then we check which inputs are required for the algorithm using the requirements attributes in the template 
+        
+        # Then we pass everything into the ivim_fit method which performs the fit according to the algorithm
+        
+        #args = [data, use_bvalues, use_initial_guess, use_bounds, use_thresholds]
+        args = [arg for arg in args if arg is not None]
+        results = self.ivim_fit(*args, **kwargs)
+        
+        #self.parameter_estimates = self.ivim_fit(data, bvalues)
+        return results
+    
+    def osipi_print_requirements(self):
```

Comment:
Take it or leave it, but this looks a lot like a `__str__` function. Maybe having that would be useful too?

Reply:
Could add that in the next iteration. I'm personally a fan of self explanatory function names, in comparison to just having str(OsipiBase).